static/app/views/performance/transactionSummary/transactionOverview/vitalsChart/content.tsx

```diff
@@ -28,7 +28,7 @@ type Props = {
     start: number;
   };
 } & Omit<React.ComponentProps<typeof ReleaseSeries>, 'children' | 'queryExtra'> &
-  Pick<React.ComponentProps<typeof LineChart>, 'onLegendSelectChanged' | 'legend'>;
+  Pick<LineChartProps, 'onLegendSelectChanged' | 'legend'>;
```

Comment:
Meh, I'm always on the fence for these :D like I wouldnt mind if you just added them to props with something like.
```
onLegendSelectChanged: LinechartProps["onLegendSelectChanged"],
legend: LinechartProps["legend"]
```

It tends to look a bit nicer if/when we extend the prop interface as it breaks into a separate line

```
interface Props 
   extends Pick<LineChartProps, 'onLegendSelectChanged' | 'legend'> {
   ...
}
```
We cant extend yet, because of & Omit<React.ComponentProps<typeof ReleaseSeries>, 'children' | 'queryExtra'> + this is a stylistic comment at this point so pls ignore, just wanted to leave my thought on this :)

Reply:
sure, I agree with you, but I my intentions were to do less modifications as possible for it to be a function component